hypervisor/boot/sbl/seed_parse.c

```diff
@@ -67,3 +68,42 @@ void parse_seed_list(struct seed_list_hob *seed_hob)
 	trusty_set_dseed(NULL, 0U);
 	(void)memset(dseed_list, 0U, sizeof(dseed_list));
 }
+
+void parse_seed_list_abl(void *boot_params, uint8_t *serial, uint32_t serial_len)
+{
+	uint32_t i;
+	struct seed_info dseed_list[BOOTLOADER_SEED_MAX_ENTRIES];
+	struct dev_sec_info *sec_info = (struct dev_sec_info *)boot_params;
+
+	if (sec_info == NULL)
+		goto fail;
+
+	(void)memset(dseed_list, 0U, sizeof(dseed_list));
+	for (i = 0U; i < sec_info->num_seeds; i++) {
+		dseed_list[i].cse_svn = sec_info->seed_list[i].svn;
+		(void)memcpy_s(dseed_list[i].seed,
+				sizeof(dseed_list[i].seed),
+				sec_info->seed_list[i].seed,
+				sizeof(sec_info->seed_list[i].seed));
+
+		/* replace original seeds with new seeds(derived with emmc serial) */
+		if (hkdf_sha256(sec_info->seed_list[i].seed,
```

Comment:
i'm thinking if we can use the legacy seed (fixed seed) to derive the RPMB key. then we don't need to derive, but instead sending the last seed legacy seed to SOS, then sos kernel (rpmb raw access module) will derive the RPMB key (together with the serial #). can you check with Yang to see if kernel driver can get the serial #)?